kafka-client/src/main/java/dev/responsive/kafka/api/ResponsiveKeyValueParams.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.api;
+
+import dev.responsive.kafka.store.SchemaType;
+import dev.responsive.utils.TableName;
+import java.time.Duration;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+public final class ResponsiveKeyValueParams {
+
+  private final TableName name;
+  private final SchemaType schemaType;
+
+  private boolean timestamped;
+  @Nullable private Duration timeToLive;
+
+  private ResponsiveKeyValueParams(
+      final String name,
+      final boolean timestamped,
+      final SchemaType schemaType
+  ) {
+    this.name = new TableName(name);
+    this.timestamped = timestamped;
+    this.schemaType = schemaType;
+  }
+
+  public static ResponsiveKeyValueParams fact(final String name) {
+    return new ResponsiveKeyValueParams(name, false, SchemaType.FACT);
+  }
+
+  public static ResponsiveKeyValueParams timestampedFact(final String name) {
+    return new ResponsiveKeyValueParams(name, true, SchemaType.FACT);
+  }
+
+  public static ResponsiveKeyValueParams keyValue(final String name) {
+    return new ResponsiveKeyValueParams(name, false, SchemaType.KEY_VALUE);
+  }
+
+  public static ResponsiveKeyValueParams timestamped(final String name) {
+    return new ResponsiveKeyValueParams(name, true, SchemaType.KEY_VALUE);
+  }
+
+  public ResponsiveKeyValueParams timestamped() {
+    this.timestamped = true;
```

Comment:
If we already have distinct static constructors for the timestamped variants, I'd rather not have a separate setter API -- can we make `timestamped` final` and remove this?